app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+                    'categories' => [$post->category],
+                    'post_excerpt' => $post->description,
+                    'post_id' => $post->target_post_id,
+                    'post_status' => 'draft',
+                    'feature_img' => $post->feature_img,
+                    'author_name' => $post->author_name,
+                    'cback_url' => action('Webapp\\PostController@getEdit', $post->id),
+                    'priority' => 0
+                ]))->send('edit')->getResult();
+                $post->content = Post::replaceIDButtonLink($returnData->data->post_id, $post->content, $post->target_post_id);
+                $post->target_post_id = $returnData->data->post_id;
+            }
+
+            $post->published_status = 0;
+            $this->_flashMessage($request, 'OK', 'success');
+        } catch(\Exception $e) {
+            $post->content = Post::replaceIDButtonLink('__POSTID__', $post->content, $post->target_post_id);
+            $this->activityLog->setInfo('Error when reaching to target site');
+            $this->activityLog->addMeta('FailedPostToTargetSite', 'error when reaching to target site');
+            $this->_flashMessage($request, '更新に失敗しました。もう一度ご確認ください。', 'warning');
+        }
+
+        $post->save();
+
+        if($request->get('hasFilter', false) !== false) {
+            return redirect()->back()->send();
+        }
+
+
+        $this->activityLog->addMeta('NewPublishedStatus', $post->published_status);
+
+        if(!$request->get('selected_ids'))
+        return redirect()->to(action('Webapp\\PostController@getIndex'))->send();
+    }
+
+    public function postRequestNewId() {
+        $post = new Post();
+        $post->user_id = $this->currentUser->id;
+        $post->save();
+
+        $this->activityLog->setPostId($post->id);
+
+        return ['id' => $post->id];
+    }
+
+    public function postAutosave(Request $request) {
+        $postId = $request->get('post_id', 0);
+        $postContent = $request->get('post_content', '');
+        $postTitle = $request->get('post_title', '');
+        $postDescription = $request->get('post_description', '');
+        $checksum = $request->get('checksum', 0);
+        $word_count = (int)$request->get('word_count', 0);
+
+        $autoSaveInterval = 4;
+
+        if($postId == 0 && empty($postContent) && empty($postTitle) && empty($postDescription)) {
+            return ['id' => 0];
+        }
+
+        if($request->session()->get('autoSave'.$postId) === null) {
+            $request->session()->set('autoSave'.$postId, [
+                'count' => 0,
+                'currentContent' => null
+            ]);
+        }
+
+        $autoSaver = $request->session()->get('autoSave'.$postId);
+
+        $calculatedChecksum = Post::calculateChecksum($postContent);
+
+        /** @var Post $post */
+        $post = Post::find($postId);
+
+        if(empty($post)) {
+            return ['error' => [
+                'message' => 'Post not found',
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Post not found" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS1192)